pkg/dispatcher/noop_dispatcher.go

```diff
@@ -28,6 +29,36 @@ func NewNoopDispatcher(consumerDao dao.ConsumerDao, sourceClient cloudevents.Sou
 
 // Start is a no-op implementation.
 func (d *NoopDispatcher) Start(ctx context.Context) {
+	// handle client reconnected signal and resync status from consumers for this source
+	d.resyncOnReconnect(ctx)
+}
+
+// resyncOnReconnect listens for client reconnected signal and resyncs all consumers for this source.
+func (d *NoopDispatcher) resyncOnReconnect(ctx context.Context) {
+	log := logger.NewOCMLogger(ctx)
+	// receive client reconnect signal and resync current consumers for this source
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-d.sourceClient.ReconnectedChan():
+			// when receiving a client reconnected signal, we resync all consumers for this source
+			// TODO: optimize this to only resync resource status for necessary consumers
+			consumerIDs := []string{}
+			consumers, err := d.consumerDao.All(ctx)
+			if err != nil {
+				log.Error(fmt.Sprintf("failed to get all consumers: %v", err))
+				continue
+			}
+
+			for _, c := range consumers {
+				consumerIDs = append(consumerIDs, c.ID)
+			}
+			if err := d.sourceClient.Resync(ctx, consumerIDs); err != nil {
```

Comment:
do we concern about the duplicated works here?

Reply:
what do you mean by duplicated works?

Reply:
after discussed offline, it should be OK since:
1. resync response will be distributed to the certain maestro instance. 
2. agent will not send the resync status repeatedly if the resync status is sent once.